corehq/apps/change_feed/producer/postgres.py

```diff
@@ -0,0 +1,61 @@
+import select
+import json
+import psycopg2.extensions
+from kafka import KeyedProducer
+from pillowtop.feed.interface import ChangeMeta
+
+from ..data_sources import get_document_store, POSTGRES
+from ..utils import send_to_kafka
+
+CHANNELS = {
+    XFormChannel.name: XFormChannel
+}
+
+
+class PostgresProducer(object):
+    """
+    Producer that pushes changes to Kafka
+    """
+
+    def __init__(self, kafka, data_source_type, data_source_name):
+        self._kafka = kafka
+        self._producer = KeyedProducer(self._kafka)
+        self._conn = get_document_store(POSTGRES)
+        self._conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        self._data_source_type = data_source_type
+        self._channel = CHANNELS[data_source_name]()
+
+    def listen(self):
+        # Each instance of PostgresProducer will take up one connection to the master database. Keep these to
+        # a minimum
```

Comment:
sorry, no i meant one "pg listener" per shard that publishes to kafka. i think we need an app layer